R/Backend-class.R

```diff
@@ -240,13 +217,16 @@ setMethod(
 #'
 #' @inheritParams backendInitialize
 #' @param file The path to the source (generally .mzML) file.
+#' @param processingQueue `list` of `ProcessingStep` objects defining the
+#'     processing steps to be applied to the spectra before returning them.
 #' @return A list of [Spectrum-class] objects.
 #' @family Backend generics
 #' @author Sebastian Gibb \email{mail@@sebastiangibb.de}
 #' @noRd
 setGeneric(
     "backendReadSpectra",
-    def=function(object, file, spectraData, ..., BPPARAM=bpparam())
+    def=function(object, spectraData, processingQueue = list(), ...,
+                 BPPARAM=bpparam())
```

Comment:
I thought adding `BPPARAM` to all methods was a good idea but now we revert the design and `MSnExperiment,spectrapply` is doing the parallelisation. So maybe we should remove `BPPARAM` from `backendReadSpectra`/`backendWriteSpectra`. (I am not familiar with nested parallelisation.)

I would keep it for `backendImportData` to speed up backend creation.

What do you think is the best way?

Reply:
nested paralellization: very bad thing. Yes, we should kick the parameter out of all `backend...` methods (except `backendImportData`).